### DISCUSSION TITLE
Lesson7 makefile commands dosen't work

### DISCUSSION THREAD

**Discussion by RaferGuo** [Cyfrin/foundry-full-course-cu#759] [Q&A]
at 2:15 nearby, when i use make deploy or make deploy-sepolia, i got error below.
i copy ether api key from
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/67697e22-810b-4289-acb9-ddae409041b7)
i copy there code from your repo
```
-include .env

.PHONY: all test clean deploy fund help install snapshot format anvil 

DEFAULT_ANVIL_KEY := 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80

help:
	@echo "Usage:"
	@echo "  make deploy [ARGS=...]\n    example: make deploy ARGS=\"--network sepolia\""
	@echo ""
	@echo "  make fund [ARGS=...]\n    example: make deploy ARGS=\"--network sepolia\""

all: clean remove install update build

# Clean the repo
clean  :; forge clean

# Remove modules
remove :; rm -rf .gitmodules && rm -rf .git/modules/* && rm -rf lib && touch .gitmodules && git add . && git commit -m "modules"

install :; forge install cyfrin/foundry-devops@0.0.11 --no-commit && forge install smartcontractkit/chainlink-brownie-contracts@0.6.1 --no-commit && forge install foundry-rs/forge-std@v1.5.3 --no-commit

# Update Dependencies
update:; forge update

build:; forge build

test :; forge test 

snapshot :; forge snapshot

format :; forge fmt

anvil :; anvil -m 'test test test test test test test test test test test junk' --steps-tracing --block-time 1

NETWORK_ARGS := --rpc-url http://localhost:8545 --private-key $(DEFAULT_ANVIL_KEY) --broadcast

ifeq ($(findstring --network goerli,$(ARGS)),--network geroli)
	NETWORK_ARGS := --rpc-url $(GEROLI_RPC_URL) --private-key $(PRIVATE_KEY) --broadcast --verify --etherscan-api-key $(ETHERSCAN_API_KEY) -vvvv
endif

deploy:
	@forge script script/DeployFundMe.s.sol:DeployFundMe $(NETWORK_ARGS)

fund:
	@forge script script/Interactions.s.sol:FundFundMe $(NETWORK_ARGS)

withdraw:
	@forge script script/Interactions.s.sol:WithdrawFundMe $(NETWORK_ARGS)
```
and when i run meke deploy, meet this error
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/422d97ab-f909-4b63-8f05-3c9e55b335e5)
i also cant run like make deploy-sepolia
even i use alt+z to put them in one line.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/aec0eee1-7a9f-4d82-bb9a-94622c0c9594)
so how can i fix those two bugs,thx

**Comment by 0xTagger** [comment]
Please try changing it from Goerli in your .env file to Sepolia and using the `make deploy `command with Sepolia instead of Goerli.
Be sure to use `source .env` command once the information is updated before you try running the `make deploy` command.

**Reply by RaferGuo** [reply]
i change it from georli to sepolia,but it still doesnt work.and report the same.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/0ebe63cd-fdce-4bf8-be13-a4fb1ea78a65)

**Reply by AztecWarlord** [reply]
you should not post your ```API_KEY```  or your ```PRIVATE_KEY``` Anywhere online!
Better yet Do Not post any contents from your ```.env``` files!

**Reply by RaferGuo** [reply]
thx，i have already delete .env pics.

**Comment by itznishant** [comment]
> i change it from georli to sepolia,but it still doesnt work.and report the same. ![image](https://user-images.githubusercontent.com/31519376/264355785-0ebe63cd-fdce-4bf8-be13-a4fb1ea78a65.png)

It seems your ```deploy-sepolia``` is incorrect. You're missing ```@``` at the start. Should be like this below. 

<img width="924" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/17523443/2464f093-19a5-47a7-a079-bc6968b48716">

I recommend to use single words as shortcuts like ```deploy``` only.
Also you need to run the ```make``` command as below. Example is for **sepolia** network.

``` make deploy ARGS="--network sepolia"```

<img width="1144" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/17523443/a00cad0c-1ac8-4515-9741-2d563f678925">


Also, make sure to run source .env as well.

**Reply by RaferGuo** [reply]
i found i should use tab instead space before forge.and the "forge" color becomes white.
im sure to run source .env. and i copy the Makefile from patrick's repo, but meet the probloem below.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/a70d9240-03e1-40c7-860a-7f19f21ec2de)
and all the code is in my repo.
and when i print below
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/f9c9f7de-e148-4c41-b9e0-ebeaa99993e6)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/2e449ca8-6a20-4a64-b5f5-97d9d3561df6)
and the transaction actually happen.
i still dont know how to fix it.
